webapp/channels/src/components/admin_console/team_channel_settings/channel/details/channel_moderation.tsx

```diff
@@ -21,172 +21,150 @@ import {ChannelModerationRoles} from './types';
 
 const PERIOD_TO_SLASH_REGEX = /\./g;
 
-const formattedMessages: any = defineMessages({
+const channelModerationTranslations = ({
     [Permissions.CHANNEL_MODERATED_PERMISSIONS.CREATE_POST]: {
-        title: {
+        title: defineMessage({
```

Comment:
![Uploading image.png…]()

Reply:
`defineMessages` have to be at the middle level (i.e. `[Permissions.CHANNEL_MODERATED_PERMISSIONS.CREATE_POST]: defineMessages({`)

That being said, you have to improve the type of several parts. First of all, you should update the constant, so the type is a const, and not a string. (i.e. constants/permissions.ts => `CREATE_POST: 'create_post' as const,`, etc...)

Then, improve the type of the name prop to something like `name: typeof Permissions.CHANNEL_MODERATED_PERMISSIONS[keyof typeof Permissions.CHANNEL_MODERATED_PERMISSIONS];`

Then in order to make sure the type is properly propagated, instead of doing `if ('something' in messages[name])` do something like `const msg = messages[name]; if ('something' in msg)`.

The next big issue is on `errorMessageToDisplay` that uses a type that we probably cannot change so easily as we did with the prop. In order to force a bit the type matching, we could do something a bit unnecessary, but would make sure everything is like it should:
`const key = Object.values(Permissions.CHANNEL_MODERATED_PERMISSIONS).find((v) => v === entry.name)`
That way you can use key as one of the proper keys, and if the key doesn't exist, handle it, instead of hard crashing with a "trying to access whatever of undefined".

Reply:
@larkox i am not sure if the updated code corresponds with the type changes you mentioned. Can you take a look if something needs to be changes?